### PR TITLE
chore: bump `plugins.json` to v2

### PR DESCRIPTION
**Pull request by ehmicky**
Fixes #260.

Before `@netlify/build@17.4.0` ([PR](https://github.com/netlify/build/pull/3361)), adding new properties to the `compatibility` field in `plugins.json` would crash. `@netlify/build@17.4.0` was added to `netlify-cli@5.4.1`. 

In order to add new properties to the `compability` field, this PR bumps the `plugins.json` format version from v1 to v2.
Versions older than `@netlify/build@17.4.0` and `netlify-cli@5.4.1` use either https://netlify-plugins.netlify.app/plugins.json or https://list-v1--netlify-plugins.netlify.app/plugins.json. Since bumping to v2 will freeze the contents of https://list-v1--netlify-plugins.netlify.app/plugins.json, those older versions will keep working.

In other words, the only difference between v1 and v2 is that it adds a new requirement for consumers: they should not fail on unknown properties in the `compatibility` field.

We will then need to update the URL in Netlify Build, CLI and front-end app to https://list-v2--netlify-plugins.netlify.app/plugins.json so they get the latest `plugins.json` changes.

https://list-v2--netlify-plugins.netlify.app/plugins.json will be automatically created/deployed by merging this PR to `main`.

### CODE REVIEW

**Review by erezrokah — APPROVED** [review]

**Comment by erezrokah** [comment]
This looks great @ehmicky.

Would it make sense to expose the URL as part of the package exports?
For example consumers could:
```
const { url, list } = require('@netlify/plugins-list') 
```

then fetch `url` or fallback to `list`?

The purpose is to avoid having to update the `url` in the consumers (it will be tied to dependencies updates).

If so I can open an issue for it. I don't think the priority for it is very high though as we have a working solution.

**Comment by ehmicky** [comment]
This makes sense.
One consequence this will have, to keep in mind: any breaking change in `plugins.json` syntax would require not only updating the syntax version (like in this PR), but also make a major release of the npm package.

**Comment by ehmicky** [comment]
Done in https://github.com/netlify/plugins/pull/407
